src/differentials.jl

```diff
@@ -205,8 +209,8 @@ add_zero(::Zero, b) = b
 add_zero(a, ::Zero) = a
 
 mul_zero(::Zero, ::Zero) = Zero()
-mul_zero(::Zero, ::Any) = Zero()
-mul_zero(::Any, ::Zero) = Zero()
+mul_zero(::Zero, x::Any) = zero(x)
+mul_zero(x::Any, ::Zero) = zero(x)
```

Comment:
Actually I might have to restrict this change to `x::Partials`.... because this change messed up `abs2` where it used to get `add(Partials(...), Zero)` it now gets `add(Partials(...), 0.0)` `mul_zero` being called with a number and then passed as the second argument.

Reply:
Or should we define `+` on `Partials` lolzz

Reply:
I think you are missing a call to `extern` somewhere in ForwardDiff2.